torch/onnx/_internal/fx/passes/shape_inference.py

```diff
@@ -45,10 +47,63 @@ def to_fake_tensor(x):
                 module.named_parameters(), module.named_buffers()
             )
         }
-
         # Shape inference via FakeTensorProp
         with stateless._reparametrize_module(module, fake_parameters_and_buffers):
             # Assign output types and shapes to each node without meta values.
-            fake_tensor_prop.FakeTensorProp(module, fake_tensor_mode).propagate(*args)
-
+            ValidationFakeProp(module, fake_tensor_mode).propagate(*args)
         return module
+
+
+@compatibility(is_backward_compatible=False)
+class ValidationFakeProp(torch.fx.Interpreter):
```

Comment:
@BowenBao @wschin . This is the solution for us, but I am not sure what stage we are with FakeTensorProp now. It seems it's been actively developed...